### PR TITLE
Reactor Netty Server feature to 5.x branch

### PR DESCRIPTION
**Pull request by Maheswari-Nagendran**
Same as https://github.com/resteasy/resteasy-spring-boot/pull/169, but targeting resteasy 5.x for apps that are not quite yet ready to move to resteasy:6.0.*.

I changed the version from `5.0.0.Final` to `5.1.0-SNAPSHOT`. Hopefully that's ok? 

@crankydillo, @PranavPMehta, @dnagarajan89 please help with review items

### CODE REVIEW

**Comment by Maheswari-Nagendran** [comment]
Please help to add branch 5.x 
This is not targeted to main branch

**Comment by liweinan** [comment]
@Maheswari-Nagendran Is there specific product requirement to put the new feature into older branches? The common criteria is that unless there is a product/project do needs this we usually keep new feature in main branch.

**Comment by jamezp** [comment]
Also note that features should not go into a micro-release without a very good reason.

**Comment by crankydillo** [comment]
Let me try to give a little clarity to this PR (and #180).  Our team contributed [the reactor-netty server adapter](https://github.com/resteasy/resteasy/pull/2693) in 2021.  We built that and the spring-boot starter pieces at the same time using internal forks.  We were unable to contribute the starter piece at that time, so we are still using an internal fork of this project and are trying to rectify that.  However, we don't think we can upgrade to Resteasy 6 in the near term, so we can't leverage #169.  We do hope to be on Resteasy 5 some time soon.  

Long story short, we're trying to get off an internal fork of this project, which contains this not-new-to-us feature.  I totally understand the argument that we should pay a price for this delayed contribution:P  Even so, we are hoping for some mercy!

**Comment by jamezp** [comment]
This does make sense. It's up to @liweinan, but it seems one option would be a 5.1 version if you are using the 5.x branches.

As for the 6.x branches, that makes sense. The javax -> jakarta migration is not simple. There are some utilities that can help, however, I do understand it's not that simple.

**Comment by Maheswari-Nagendran** [comment]
Targeted the changes to 5.1 version
CC: @liweinan, @jamezp, @crankydillo, @PranavPMehta, @dnagarajan89 
Please help review

**Comment by liweinan** [comment]
@jamezp Thanks for checking! @crankydillo Ok I understand the requirement, I'll do a `5.1` release accordingly.

**Comment by liweinan** [comment]
@Maheswari-Nagendran I run your branch locally and the enforcer check failed:

image

Could you please cleanup these issues? It seems the Github CI doesn't cover the build/test failures.

**Comment by Maheswari-Nagendran** [comment]
> @Maheswari-Nagendran I run your branch locally and the enforcer check failed:
> 
> image
> 
> Could you please cleanup these issues? It seems the Github CI doesn't cover the build/test failures.

@liweinan - when we run `mvn clean install` from local we are not getting this error and tried to configure [this plugin](https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html) locally but still did not face the warning which you showed (instead i could see in some 2 dependency jars only)

Can you please provide the command you run which showed these enforcer issues?

**Comment by liweinan** [comment]
@Maheswari-Nagendran Here is the branch I used:

![CB40069C-D8EE-4200-8DB8-C236A84A32E0](https://user-images.githubusercontent.com/201907/189162383-3e4da8e0-8c96-43f1-8f42-a23905e2e3c9.png)

Then I run the following command:

```bash
➤ mvn clean install
```

And here is the result I get:

![A07825F8-869C-4A5D-963C-DC25FAB92CA6](https://user-images.githubusercontent.com/201907/189162523-7b64334c-d11d-43b1-a21d-188daf955f77.png)

![FF399609-CD34-4063-B3CF-A4722CB6ED57](https://user-images.githubusercontent.com/201907/189162553-3b63380b-81d0-47ef-be84-f257c14c25d1.png)

Here are the Maven and Java version I used(which IMHO should not be relevant with enforcer check rule error):

```bash
weli@ovpn-12-50:~/w/resteasy-spring-boot|5.x-reactor-netty-server⚡?
➤ java -version                                                                                                                                                                                                                                                                                                 23:24:48
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)
weli@ovpn-12-50:~/w/resteasy-spring-boot|5.x-reactor-netty-server⚡?
```

```bash
➤ mvn -version                                                                                                                                                                                                                                                                                                  23:25:07
Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)
Maven home: /Users/weli/.sdkman/candidates/maven/current
Java version: 17.0.2, vendor: Oracle Corporation, runtime: /Users/weli/.sdkman/candidates/java/17.0.2-open
Default locale: zh_CN_#Hans, platform encoding: UTF-8
OS name: "mac os x", version: "12.3.1", arch: "x86_64", family: "mac"
```

You may need to exclude conflicting dependencies from `pom.xml`  and define the version accurately (which I also need to do for the `main` branch).

Here is example to do so: https://github.com/resteasy/resteasy-spring-boot/blob/5.x/resteasy-spring-boot-starter/pom.xml#L81

**Comment by Maheswari-Nagendran** [comment]
@liweinan - Thanks for the guidance
Added the exclusions as appropriate. Please help review the same and let me know if there are any more issues on it

**Comment by Maheswari-Nagendran** [comment]
@liweinan - even with the suggestions you have provided [here](https://github.com/resteasy/resteasy-spring-boot/pull/184#issuecomment-1240877006), I was not able to get `spring-core`, `spring-webmvc` and `spring-test` dependency duplication problem. Hence I missed those in previous commit
Again I rerun `mvn clean install` and did not see any issues.
Please let us know if you face any further issues in it. Thanks for your support

**Comment by liweinan** [comment]
@Maheswari-Nagendran Thanks for the fix! In addition the test engine needs to be added for the test to pass locally in my Java 11 environment. I created a PR to your branch, and could you please merge it:

- https://github.com/Maheswari-Nagendran/resteasy-spring-boot/pull/1/files

**Comment by liweinan** [comment]
In addition, as the PR is almost ready for merge, @Maheswari-Nagendran We require all contributions to be made under the terms of the ASL 2.0 License:[ http://www.apache.org/licenses/LICENSE-2.0](http://www.apache.org/licenses/LICENSE-2.0) 
Do you agree to these terms?

**Comment by Maheswari-Nagendran** [comment]
> @Maheswari-Nagendran Thanks for the fix! In addition the test engine needs to be added for the test to pass locally in my Java 11 environment. I created a PR to your branch, and could you please merge it:
> 
> * https://github.com/Maheswari-Nagendran/resteasy-spring-boot/pull/1/files

Yea merged your changes to my fork branch
Thanks @liweinan

**Comment by Maheswari-Nagendran** [comment]
> In addition, as the PR is almost ready for merge, @Maheswari-Nagendran We require all contributions to be made under the terms of the ASL 2.0 License:[ http://www.apache.org/licenses/LICENSE-2.0](http://www.apache.org/licenses/LICENSE-2.0) Do you agree to these terms?

Yes I agree.
PLease help merge this and release 5.1.0 version of resteasy

**Comment by liweinan** [comment]
@Maheswari-Nagendran Thanks! I'll do the release accordingly.

**Comment by Maheswari-Nagendran** [comment]
> @Maheswari-Nagendran Thanks! I'll do the release accordingly.

Thanks @liweinan

**Comment by liweinan** [comment]
`5.1.0.Final` is released: 

- [Release v5.1.0.Final · resteasy/resteasy-spring-boot · GitHub](https://github.com/resteasy/resteasy-spring-boot/releases/tag/5.1.0.Final)

JARs are uploaded to JBoss repo and should be synced to Maven central soon.

**Comment by Maheswari-Nagendran** [comment]
> JBoss

@liweinan - SInce this was not published to Maven central yet can you please share the JBOSS repo where you published the jars

**Comment by liweinan** [comment]
@Maheswari-Nagendran It's weird that the release is not published in JBoss repo yet:

https://repository.jboss.org/nexus/content/repositories/releases/org/jboss/resteasy/resteasy-spring-boot-starter/

It may caused by cache problems or something else. I'll check it and if it is still not published I'll do publish again.

**Comment by liweinan** [comment]
I can find the release is already public:

https://repository.jboss.org/nexus/content/groups/public/org/jboss/resteasy/resteasy-spring-boot-starter-common/5.1.0.Final/

But I'm not sure why it's not synced with jboss repo and Maven central repo yet. Maybe it needs some time. If it's still not synced with repos yet by this weekend, I'll report problem to our engineering team.

**Comment by liweinan** [comment]
It's synced to Maven Central Repo: https://search.maven.org/artifact/org.jboss.resteasy/resteasy-spring-boot-starter-common/5.1.0.Final/jar

**Comment by Maheswari-Nagendran** [comment]
> https://search.maven.org/artifact/org.jboss.resteasy/resteasy-spring-boot-starter-common/5.1.0.Final/jar

Thanks for the support and update
